provoke/worker/__init__.py

```diff
@@ -113,17 +113,61 @@ class RequeueTask(Exception):
     pass
 
 
-class _WorkerProcess(object):
+def _run_cb(obj, name, *args, **kwargs):
+    try:
+        callback = getattr(obj, name)
+    except AttributeError:
+        return
+    try:
+        callback(*args, **kwargs)
+    except (DiscardTask, AssertionError, SystemExit, KeyboardInterrupt):
+        raise
+    except Exception:
+        logger.exception('Unhandled exception in callback')
```

Comment:
I didn't get a chance to look at this before it was merged, but wouldn't this keep any exceptions besides `DiscardTask`, `AssertionError`, `SystemExit`, or `KeyboardInterrupts` from propagating to the caller?